Dockerfile.windows

```diff
@@ -152,15 +153,14 @@
 # -----------------------------------------------------------------------------------------
 
 
+# Extremely important. If using this Dockerfile in process isolated containers,
+# the kernel of the host must match the container image, and hence would fail
+# between Windows Server 2016 (aka RS1) and Windows Server 2019 (aka RS5).
+ARG WINDOWS_VERSION=${WINDOWS_VERSION:-1809}
+
 # The number of build steps below are explicitly minimised to improve performance.
 
-# Extremely important - do not change the following line to reference a "specific" image, 
-# such as `mcr.microsoft.com/windows/servercore:ltsc2019`. If using this Dockerfile in process
-# isolated containers, the kernel of the host must match the container image, and hence
-# would fail between Windows Server 2016 (aka RS1) and Windows Server 2019 (aka RS5).
-# It is expected that the image `microsoft/windowsservercore:latest` is present, and matches
-# the hosts kernel version before doing a build. 
-FROM microsoft/windowsservercore
+FROM mcr.microsoft.com/windows/servercore:${WINDOWS_VERSION}
```

Comment:
I'm not sure I like this. I think a more consistent approach that would work uniformly is (actually what mostly happens today) to have a known image tag pre-loaded (ie `microsoft/windowsservercore[:latest]`) which IS the matching build from host to container, regardless of OS.

Reply:
That works for CI, but it would require users to "spoof" the `microsoft/windowsservercore` image before being able to build the Dockerfile. That'd be a big no-no to me.

It'd be like `docker tag busybox ubuntu:latest` `docker build FROM ubuntu`

Reply:
Another idea would be, can we have a manifest list `mcr.microsoft.com/windows/servercore` that just points to the right OS version specific image? If there is this, then it would help here. I doubt this will happen as recently the latest tag was removed as announced months ago.

Otherwise we should really think of moving away from the old names and work with the official image names that MSFT gives us.

Reply:
>That works for CI, but it would require users to "spoof" the microsoft/windowsservercore image before being able to build the Dockerfile. That'd be a big no-no to me.

There's no "spoofing" going on here. It's simply tagging the right servercore image with a well known consistent name.

Reply:
The repo name is a tad offensive, but if you want to play with how well Docker (or other tools) handle a manifest list of this variety, I've got one I've been playing with as a test of other things up at https://hub.docker.com/r/hell/win (image bits still come from Microsoft, of course).